File: src/Shithead/Core/Hand.cs
using System.Collections.Generic;

namespace Shithead.Core
{
    public class Hand
    {
        public List<Card> FaceDownCards { get; internal set; }
        public List<Card> FaceUpCards { get; internal set; }
        public List<Card> InHandCards { get; internal set; }

        public Hand()
        {
            FaceDownCards = new List<Card>(3);
            FaceUpCards = new List<Card>(3);
            InHandCards = new List<Card>();
        }

        public Hand(List<Card> faceDownCards, List<Card> faceUpCards, List<Card> inHandCards)
        {
            FaceDownCards = faceDownCards;
            FaceUpCards = faceUpCards;
            InHandCards = inHandCards;
        }
    }
}

File: src/Shithead/WebUI/js/game/modules/hand.js
var Hand = function (handData, gameHub) {

    var self = this;

    this.gameHub = gameHub;
    
    this.faceDownCards = handData.FaceDownCards.map(function (card, i) {
        return new Card(card.Rank, card.Suit, false, self);
    });

    this.faceUpCards = handData.FaceUpCards.map(function (card, i) {
        return new Card(card.Rank, card.Suit, true, self);
    });

    this.inHandCards = handData.InHandCards.map(function (card, i) {
        return new Card(card.Rank, card.Suit, true, self);
    });
};

Hand.prototype = {

    constructor: Hand,

    draw: function (container) {

        this.faceDownCards.forEach(function (card, i) {
            card.draw(container);
        });

        this.faceUpCards.forEach(function (card, i) {
            card.draw(container);
        });

        this.inHandCards.forEach(function (card, i) {
            card.draw(container);
        });
    },

    setPosition: function (playerIndex, container, layoutParams) {

        container.x = layoutParams[playerIndex].x;
        container.y = layoutParams[playerIndex].y;
        container.rotation = layoutParams[playerIndex].rot;

        this.faceDownCards.forEach(function (card, i) {
            card.setPosition(i * 70, 0);
        });
        this.faceUpCards.forEach(function (card, i) {
            card.setPosition(i * 70, 60);
        });
        this.inHandCards.forEach(function (card, i) {
            card.setPosition(i * 20, 80);
        });
    }
};